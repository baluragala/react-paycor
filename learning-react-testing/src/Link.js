import React, { Component } from "react";

class Link extends Component {
  constructor(props) {
    super(props);
    this.state = { clicked: false };
  }

  handleClick = () => {
    window.alert("clicked");
    this.setState({ clicked: true });
  };

  render() {
    let { url, title } = this.props;
    return (
      <a href={url} onClick={this.handleClick}>
        {title}
      </a>
    );
  }
}

export default Link;
