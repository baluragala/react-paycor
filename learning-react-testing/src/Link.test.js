import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallowToJson } from "enzyme-to-json";
import Link from "./Link";

//React 16
Enzyme.configure({ adapter: new Adapter() });

describe("Link", () => {
  it("should render correctly", () => {
    const wrapper1 = shallow(<Link title="tester" url="https://test.com" />);
    expect(shallowToJson(wrapper1)).toMatchSnapshot();
    const wrapper2 = shallow(<Link title="test" url="https://test.com" />);
    expect(shallowToJson(wrapper2)).toMatchSnapshot();
  });

  it("should handle the click event", () => {
    window.alert = jest.fn();
    const wrapper = shallow(<Link title="tester" url="https://test.com" />);
    wrapper.simulate("click");
    expect(window.alert).toHaveBeenCalledWith("clicked");
  });

  it("should have rendered props correctly", () => {
    const wrapper = shallow(<Link title="tester" url="https://test.com" />);
    expect(wrapper.props().href).toEqual("https://test.com");
    expect(wrapper.props().children).toEqual("tester");
  });

  it("should handle state changes", () => {
    const wrapper = shallow(<Link title="tester" url="https://test.com" />);
    expect(wrapper.state().clicked).toBeFalsy();
    wrapper.simulate("click");
    expect(wrapper.state().clicked).toBeTruthy();
  });
});
