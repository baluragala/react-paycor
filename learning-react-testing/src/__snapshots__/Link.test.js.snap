// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly 1`] = `
<a
  href="https://test.com"
  onClick={[Function]}
>
  tester
</a>
`;

exports[`Link should render correctly 2`] = `
<a
  href="https://test.com"
  onClick={[Function]}
>
  test
</a>
`;
